File: frontend/src/lib/assetQA.ts
const BASE = "/api/asset-qa";

export interface AssetQAIssue {
  id: string;
  severity: "info" | "warning" | "critical";
  category: string;
  title: string;
  description: string;
  suggestion: string;
  detected_at: string;
}

export interface AssetQAReport {
  asset_id: string;
  score: number;
  status: "healthy" | "needs_work" | "problematic";
  issues: AssetQAIssue[];
  strengths: string[];
  recommendations: string[];
  generated_at: string;
}

export function qaStatusColor(status: string): string {
  if (status === "healthy") return "emerald";
  if (status === "problematic") return "red";
  return "amber";
}

export function qaStatusLabel(status: string): string {
  if (status === "healthy") return "HEALTHY";
  if (status === "problematic") return "PROBLEMATIC";
  return "NEEDS WORK";
}

export function severityColor(severity: string): string {
  if (severity === "critical") return "red";
  if (severity === "warning") return "amber";
  return "blue";
}

export async function runAssetQA(assetId: string): Promise<AssetQAReport> {
  const res = await fetch(`${BASE}/run/${assetId}`, { method: "POST" });
  if (!res.ok) {
    const body = await res.json().catch(() => ({}));
    throw new Error(body.detail ?? `QA analysis failed (${res.status})`);
  }
  return res.json();
}

export async function getAssetQA(assetId: string): Promise<AssetQAReport | null> {
  const res = await fetch(`${BASE}/${assetId}`);
  if (res.status === 404) return null;
  if (!res.ok) throw new Error(`Get QA report failed (${res.status})`);
  return res.json();
}

File: backend/app/models/asset_qa.py
from __future__ import annotations
from pydantic import BaseModel

VALID_QA_SEVERITIES = {"info", "warning", "critical"}
VALID_QA_CATEGORIES = {"geometry", "materials", "textures", "uv", "scale", "rigging", "export"}
VALID_QA_STATUSES = {"healthy", "needs_work", "problematic"}

# Score thresholds
SCORE_HEALTHY = 85
SCORE_NEEDS_WORK = 60


class AssetQAIssue(BaseModel):
    id: str
    severity: str       # info | warning | critical
    category: str       # geometry | materials | textures | uv | scale | rigging | export
    title: str
    description: str
    suggestion: str
    detected_at: str


class AssetQAReport(BaseModel):
    asset_id: str
    score: int = 100    # 0–100
    status: str = "needs_work"   # healthy | needs_work | problematic
    issues: list[AssetQAIssue] = []
    strengths: list[str] = []
    recommendations: list[str] = []
    generated_at: str
